lazyflow/utility/io_util/RESTfulPrecomputedChunkedVolume.py

```diff
@@ -104,30 +106,37 @@ def _init_config(self, volume_url=None, volume_description=None):
 
         jsonschema.validate(self._json_info, self.info_schema)
 
-        # Precomputed spec requires scales to be ordered from highest resolution (original) to lowest (most downscaled).
-        # We want the opposite: to load the lowest resolution, i.e. smallest file size, first.
-        self.scales = list(reversed(self._json_info["scales"]))
+        # Scales are ordered from original to most-downscaled in Precomputed spec
+        self.lowest_scale_key = self._json_info["scales"][-1]["key"]
+        self.original_resolution_key = self._json_info["scales"][0]["key"]
+        # Reverse so that the ScaleComboBox shows the options ordered from most-downscaled to original
+        self.scales = {scale["key"]: scale for scale in reversed(self._json_info["scales"])}
         self.dtype = self._json_info["data_type"]
         self.n_channels = self._json_info["num_channels"]
 
-    def get_chunk_size(self, scale=0):
+    def get_chunk_size(self, scale=""):
+        scale = scale if scale else self.lowest_scale_key
         n_channels = self.n_channels
         block_shape = numpy.array([n_channels] + self.scales[scale]["chunk_sizes"][0][::-1])
         return block_shape
 
-    def get_resolution(self, scale=0):
+    def get_resolution(self, scale=""):
+        scale = scale if scale else self.lowest_scale_key
         resolution = numpy.array(self.scales[scale]["resolution"][::-1])
         return resolution
 
-    def get_voxel_offset(self, scale=0):
+    def get_voxel_offset(self, scale=""):
+        scale = scale if scale else self.lowest_scale_key
```

Comment:
It doesn't technically need to be an optional parameter, but it does have to support the empty string as a value for `scale`. It's to some extent just a leftover of the legacy implementation, but I also didn't find it unfitting to reflect the need to handle the empty string.

Reply:
Writing this down for posterity: The root issue is that we don't know the key until we load it, but if then we feed back the key into the scale InputSlot once we know it, that re-triggers the setup of the graph. Which errors out if done from within the first setup - and even if I implement a listener from outside that does the setting after the setup is finished, then it would still double the number of web requests and time needed to initialise the web store.

Reply:
I wouldn't bake it in that way round, but the other (so having the default constant in lazyflow - in the end it should be defined there, as the code does something special with it - in ilastik it is only used).

Reply:
Sometimes the answer is so simple and obvious, and it still doesn't occur to you 😅